### DISCUSSION TITLE
Additional CMake files and MSVC support

### DISCUSSION THREAD

**Discussion by V-Strassheim** [troglobit/mdnsd#81] [General]
Hi @troglobit , would it be possible to include CMake as an **optional** build system and additions for Microsoft's compilers to make it as easy as possible to switch from a [fork](https://github.com/Pro/mdnsd) to this original project. You have already integrated some of the [changes](https://github.com/troglobit/mdnsd/pull/1) a few years ago.
As you probably already know, mdsnd is used in the [open62541](https://github.com/open62541/open62541) project. Currently a switch to mainline is considered there in context of IPv6 enhancement. Of course the work should not be left to you, I and certainly the community of open62541 are interested in getting and keeping the additional functionality stable. If you generally do not want it, I would discuss further options in open62541.

Some background: 
open62541 is an open source implementation of the open communication standard OPC UA. In the future, OPC UA could ensure better interoperability between devices, especially in the industrial environments, by replacing proprietary solutions. Therefore, the software should be usable for as many systems and users as possible. At Linutronix, we support partners from the industry in the use of open source software and are therefore actively involved in a large number of projects.

**Comment by troglobit** [comment]
Hi!  I did not know the open62541 project used mdnsd, great to hear! :smiley: 

However, I'm not too keen on pulling in another build system, or hosting support for compilers I don't use or cannot even test myself.  The reasons are several, but from previous experience, what usually happens is that it ends up as a code dump that I'm suddenly responsible for and have to maintain and support for free.  I have no interest in spending my free time on supporting users that find bugs in or   fail to use CMake (which I really dislike) or MSVC.

**Comment by V-Strassheim** [comment]
I can understand that, but it was still worth a try :smiley:  There will surely be another clean solution for open62541.
However, I would like to take this opportunity to thank you for maintaining the library and other useful tools! :+1:
